pkg/arvo/app/chat-cli.hoon

```diff
@@ -0,0 +1,1111 @@
+::  chat-cli: cli chat client using chat-store and friends
+::
+::    pulls all known messages into a single stream.
+::    type ;help for usage instructions.
+::
+::    note that while the chat-store only cares about paths,
+::    we mostly deal with [ship path] (aka target) here.
+::    when sending messages (through the chat hook),
+::    we concat the ship onto the head of the path,
+::    and trust it to take care of the rest.
+::
+/-  sole-sur=sole, *chat-store, *chat-view, *chat-hook
+/+  sole-lib=sole
+/=  seed  /~  !>(.)
+::
+|%
++$  state
+  $:  grams=(list mail)                             ::  all messages
+      known=(set [target serial])                   ::  known message lookup
+      count=@ud                                     ::  (lent grams)
```

Comment:
Is this a universal count? Seems like it would be more useful to just have a count per mailbox...

Reply:
As per the above comment, since this is a single-stream app, we have no need for per-mailbox counts.

The real reason this exists is so that we don't have to run `(lent grams)` a bunch, which becomes troublesome once your list of messages gets bigger.

Reply:
I'm going to resolve my comments about universal counts, etc for now since this is time-sensitive, but afterwards we should have a longer discussion on the subject.